qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -538,6 +539,8 @@ def run(self, unitary, **options):
                 target,
                 preferred_direction,
             )
+        elif unitary.shape[0] == unitary.shape[1]:
+            synth_dag = circuit_to_dag(qs_decomposition(unitary))
         else:
```

Comment:
Will this ever get triggered now, won't the if at L542 mean that all unitaries will go to the `qs_decomposition() now? Should we just drop the isometry usage from the pass?

Reply:
I had asummed that we were letting "unitary" be rectangular matrices here which look like they would reach the isometry decomposition. Perhaps that shouldn't be allowed for unitary synthesis, however.

Reply:
removed isometry from the pass.